bin/FileSystem.js

```diff
@@ -0,0 +1,24 @@
+module.exports = class FileSystem {
+
+  constructor(driver) {
+    this.driver = driver;
+  }
+
+  async writeFile(data, projectRelativeDirectory, filename, reporter = console.log) {
+    const directory = await this.driver.writeDirectory(projectRelativeDirectory);
+    //  Not using path.join as want to keep this layer separate from the file system driver
+    const fullPath = directory + filename;
```

Comment:
e.g. if you were on Windows the path separator would be different, most cloud providers use still `/` for storage, etc.
There is an implicit contract that you can generate a path by concatenating the strings `directory` and `filename`, if this were to become portable there would be some interesting choices to make like returning a `Directory` Value Object from the first call or requiring it as an additional parameter in the second. I think for the moment all of that would be overkill, we are only making an assumption about paths but not about any filesystem APIs, hidden in the driver.